From ec19ab070bb107d69a89a6e433b5002003cf2fe9 Mon Sep 17 00:00:00 2001
Message-Id: <ec19ab070bb107d69a89a6e433b5002003cf2fe9.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Tue, 12 Aug 2014 20:05:50 +0100
Subject: [PATCH 04/34] FROM_UPSTREAM [VPG]: drm/i915: Suppress a WARN on
 reading an object back for a GPU hang

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
Reviewed-by: Mika Kuoppala <mika.kuoppala@intel.com>
Signed-off-by: Tomas Elf <tomas.elf@intel.com>
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: 5ae2c8db6e06637c621ee96846422265552a4eed
Upstream-as-of-tag: 3.17-rc3
Issue: GMIN-3579
Tracked-On: https://jira01.devtools.intel.com/browse/GMIN-3579
Change-Id: I95cd8d95a9727e25b2fb8ba4999bb43216ffc9e5
---
 drivers/gpu/drm/i915/i915_gpu_error.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gpu_error.c b/drivers/gpu/drm/i915/i915_gpu_error.c
index 9b9339b..20ea5e4 100644
--- a/drivers/gpu/drm/i915/i915_gpu_error.c
+++ b/drivers/gpu/drm/i915/i915_gpu_error.c
@@ -564,7 +564,10 @@ i915_error_object_create(struct drm_i915_private *dev_priv,
 	if (dst == NULL)
 		return NULL;
 
-	dst->gtt_offset = i915_gem_obj_offset(src, vm);
+	if (i915_gem_obj_bound(src, vm))
+		dst->gtt_offset = i915_gem_obj_offset(src, vm);
+	else
+		dst->gtt_offset = -1;
 
 	reloc_offset = dst->gtt_offset;
 	use_ggtt = (src->cache_level == I915_CACHE_NONE &&
-- 
1.7.9.5

